frontend/lib/utils/token-utils.ts

```diff
@@ -0,0 +1,108 @@
+import { useReadContracts } from 'wagmi';
+import { parseUnits, formatUnits } from 'viem';
+
+// ERC20 ABI fragments for reading token info
+export const erc20ABI = [
+  {
+    inputs: [],
+    name: "symbol",
+    outputs: [{ internalType: "string", name: "", type: "string" }],
+    stateMutability: "view",
+    type: "function",
+  },
+  {
+    inputs: [],
+    name: "decimals",
+    outputs: [{ internalType: "uint8", name: "", type: "uint8" }],
+    stateMutability: "view",
+    type: "function",
+  },
+] as const;
+
+// Known address constants
+export const ETH_ADDRESS = "0x0000000000000000000000000000000000000003";
+
+/**
+ * Hook to fetch token information (symbol and decimals)
+ * @param tokenAddress The address of the ERC20 token
+ * @returns Token information including symbol and decimals
+ */
+export function useTokenInfo(tokenAddress?: `0x${string}`) {
```

Comment:
Address type imported from wagmi/viem
```suggestion
export function useTokenInfo(tokenAddress?: Address) {
```